packages/patterns/src/patterns/patternMatchers.js

```diff
@@ -1258,6 +1259,177 @@ const makePatternKit = () => {
     getRankCover: () => getPassStyleCover('tagged'),
   });
 
+  /**
+   * @param {CopyArray} elements
+   * @param {Pattern} elementPatt
+   * @param {bigint} bound Must be >= 1n
+   * @param {CopyArray} [inResults]
+   * @param {CopyArray} [outResults]
+   * @param {Checker} [check]
+   * @returns {boolean}
+   */
+  const elementsHasSplit = (
+    elements,
+    elementPatt,
+    bound,
+    inResults = undefined,
+    outResults = undefined,
+    check = identChecker,
+  ) => {
+    let count = 0n;
+    // Since this feature is motivated by ERTP's use on
+    // non-fungible (`set`, `copySet`) amounts,
+    // their arrays store their elements in decending lexicographic order.
+    // But this function has to make some choice amoung equally good minimal
+    // results. It is more intuitive for the choice to be the first `bound`
+    // matching elements in ascending lexicigraphic order, rather than
+    // decending. Thus we iterate `elements` in reverse order.
+    for (let i = elements.length - 1; i >= 0; i -= 1) {
+      const element = elements[i];
+      if (count < bound) {
+        if (matches(element, elementPatt)) {
+          count += 1n;
+          inResults?.push(element);
+        } else {
+          outResults?.push(element);
+        }
+      } else if (outResults === undefined) {
+        break;
+      } else {
+        outResults.push(element);
+      }
+    }
+    return check(
+      count >= bound,
+      X`Has only ${q(count)} matches, but needs ${q(bound)}`,
+    );
+  };
+
+  /**
+   * @param {CopyArray<[Key, bigint]>} pairs
+   * @param {Pattern} elementPatt
+   * @param {bigint} bound Must be >= 1n
+   * @param {CopyArray<[Key, bigint]>} [inResults]
+   * @param {CopyArray<[Key, bigint]>} [outResults]
+   * @returns {boolean}
+   * @param {Checker} [check]
+   */
+  const pairsHasSplit = (
+    pairs,
+    elementPatt,
+    bound,
+    inResults = undefined,
+    outResults = undefined,
+    check = identChecker,
+  ) => {
+    let count = 0n;
+    // Since this feature is motivated by ERTP's use on
+    // semi-fungible (`copyBag`) amounts,
+    // their arrays store their elements in decending lexicographic order.
+    // But this function has to make some choice amoung equally good minimal
+    // results. It is more intuitive for the choice to be the first `bound`
+    // matching elements in ascending lexicigraphic order, rather than
+    // decending. Thus we iterate `pairs` in reverse order.
+    for (let i = pairs.length - 1; i >= 0n; i -= 1) {
+      const [element, num] = pairs[i];
+      const numRest = bound - count;
+      if (numRest >= 1n) {
+        if (matches(element, elementPatt)) {
+          if (num <= numRest) {
+            count += num;
+            inResults?.push([element, num]);
+          } else {
+            const numIn = num - numRest;
+            count += numIn;
```

Comment:
Shouldn't `inResults` get `numRest`, and outResults get `num - numRest`?

If this isn't clear, I can try to construct a test case. I'd start by trying to remove 2 'b's from a collection that included at least 3.

Reply:
I'd love a test case, thanks!

Reply:
Added commit `abfc1f7` with a test and a proposed fix.

Reply:
Accidentally wiped that out, but it is now restored. Done by you, Thanks!